tests/utils/test_columns_validator.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime
+from typing import Any, Sequence
+
+import pytest
+
+from snuba.utils.schemas import (
+    UUID,
+    Column,
+    ColumnSet,
+    ColumnValidator,
+    Date,
+    Float,
+    Int,
+    String,
+    UInt,
+)
+
+COLUMNS = ColumnSet(
+    [
+        Column("str_param", String()),
+        Column("uint_param", UInt(32)),
+        Column("int_param", Int(32)),
+        Column("float_param", Float(64)),
+        Column("uuid_param", UUID()),
+        Column("date_param", Date()),
+    ]
+)
+
+
+@pytest.mark.parametrize(
+    "column_name, values, is_valid",
+    [
+        # valid column types
+        pytest.param("uint_param", [1, 2, 3, 4], True),
```

Comment:
If a column's type is unsigned integer and if user passes say -17 to it, we should disallow that because unsigned integers do not have negative values. Probably good to add a test case around that.